models/base.py

```
@@ -1,16 +1,31 @@
 
 class BaseModel(ABC):
 
     @staticmethod
-    def _validate_input(texts: list[str]) -> None:
+    def _validate_input(texts: list[str], text_ids: list[str] | None, num_rows: int) -> None:
         if not isinstance(texts, list):
             raise ValueError("Input must be a list of strings.")
         if not all(isinstance(text, str) for text in texts):
             raise ValueError("All items in the input list must be strings.")
+        if text_ids and len(text_ids) != num_rows:
+            raise ValueError(f"text_ids length ({len(text_ids)}) must match number of texts ({num_rows}).")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)